### ISSUE TITLE
Bug in generated code with nested models

### ISSUE THREAD

**Issue by PtiPingouin** [gfx/Android-Orma#322]
I get the following generated code for my "Model1":
```
  @NonNull
  @Override
  public String getSelectFromTableClause() {
    return "`Model1`"+ " AS " + getEscapedTableAlias()
            + " LEFT OUTER JOIN `Model3` AS " + model2.associationSchema.model3.associationSchema.getEscapedTableAlias() + " ON " + model2.associationSchema.model3.getQualifiedName() + " = " + model2.associationSchema.model3.associationSchema.mId.getQualifiedName()
            + " LEFT OUTER JOIN `Model4` AS " + model2.associationSchema.model2.associationSchema.model3.associationSchema.model4.associationSchema.getEscapedTableAlias() + " ON " + model2.associationSchema.model2.associationSchema.model3.associationSchema.model4.getQualifiedName() + " = " + model2.associationSchema.model2.associationSchema.model3.associationSchema.model4.associationSchema.mId.getQualifiedName()
      ;
  }

```

We can see:
`model2.associationSchema.model2.associationSchema.model3`

While I believe it should be:
`model2.associationSchema.model3`

My schema is quite complexe so I'm still working on isolating the problem, but so far, nothing seems to be wrong with my models.

**Comment by PtiPingouin** [comment]
For information, all my models extend the same "BaseModel" class:

```
public abstract class BaseModel {
    @PrimaryKey
    public long id;
}
```

**Comment by PtiPingouin** [comment]
(For information, this gives me a build error. I cannot launch my project anymore...)

**Comment by gfx** [comment]
Can I see your whole models? I don't have enough information to look into.

**Comment by gfx** [comment]
I can reproduce this bugs in deeply nested models. Investigating.

**Comment by gfx** [comment]
This is fixed in v3.1.1. Can you give it a try?

**Comment by PtiPingouin** [comment]
It works. Thanks!
